### ISSUE TITLE
Unwanted error logged when loading a circuit from json

### ISSUE THREAD

**Issue by stavros11** [qiboteam/qibo#1519]
The following script
```py
import json
from qibo import Circuit, gates

c = Circuit(2)
c.add(gates.CZ(0, 1))

with open("test.json", "w") as file:
    json.dump(c.raw, file)

with open("test.json", "r") as file:
    cd = json.load(file)
    cl = Circuit(2)
    cl = c.from_dict(cd)
```
produces this output
```py
[Qibo 0.2.14|ERROR|2024-11-08 14:05:50]: Cannot use `controlled_by` method on gate <qibo.gates.gates.CZ object at 0x7f13cd1a1a60> because it is already controlled by (0,).
```
It does not fail, and the loaded circuit `cl` is usable, so I would not consider it very urgent, however it is weird that the error is printed. 
This is most likely due to some `try`-`except` in the loader.

**Comment by alecandido** [comment]
Yes, the solution should be similar to the one in https://github.com/qiboteam/qibo/pull/1505: if you know that it's going to be used in a `try ... except`, you should avoid usage of `raise_error()` in favor of a plain `raise`.

`raise_error()` will first log, then raise. So, if it already logged, it is impossible to take it back, even if the error gets subsequently caught.
